presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -956,6 +956,32 @@ public void testArrayConstructor()
                 "array_constructor((bound_long + 0), (unbound_long + 1), NULL)");
     }
 
+    @Test
+    public void testRowConstructor()
+    {
+        optimize("ROW(NULL)");
+        optimize("ROW(1)");
+        optimize("ROW(unbound_long + 0)");
+        optimize("ROW(unbound_long + unbound_long2, unbound_string, unbound_double)");
+        optimize("ROW(unbound_boolean, FALSE, ARRAY[unbound_long, unbound_long2], unbound_null_string, unbound_interval)");
+        optimize("ARRAY [ROW(unbound_string, unbound_double), ROW(unbound_string, 0.0)]");
+        optimize("ARRAY [ROW('string', unbound_double), ROW('string', bound_double)]");
+        optimize("ROW(ROW(NULL), ROW(ROW(ROW(ROW('rowception')))))");
+        optimize("ROW(unbound_string, bound_string)");
+
+        // the following tests should how when the CAST to ROW is working
+
+        // these two tests don't work because one type is varchar(17) and the other varchar
+        // optimize("ARRAY [ROW(unbound_string, unbound_double), ROW(bound_string, 0.0)]");
+        // optimize("ARRAY [ROW(bound_string, 0.0), ROW(unbound_string, unbound_double)]");
+
+        // these two tests don't work because the CAST from NULL generate a row with uppercase
+        // field names (row<varchar,double>('FIELD0','FIELD1')), while the row expression
```

Comment:
Even if you don't pass any name, it's populating them :/
See [TypeSignature.java] (https://github.com/prestodb/presto/blob/c7bf63a2138c2e61f0edb307cefbf6b35fba25c2/presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java#L240)

That's why we also have some problems with fields names and lower/upper case.